custom/src/main/java/com/splunk/opentelemetry/metrics/MicrometerInstaller.java

```diff
@@ -0,0 +1,58 @@
+package com.splunk.opentelemetry.metrics;
+
+import com.google.auto.service.AutoService;
+import com.signalfx.metrics.errorhandler.OnSendErrorHandler;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.binder.jvm.ClassLoaderMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmGcMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmHeapPressureMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmMemoryMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmThreadMetrics;
+import io.micrometer.core.instrument.config.NamingConvention;
+import io.micrometer.signalfx.SignalFxMeterRegistry;
+import io.opentelemetry.javaagent.spi.ComponentInstaller;
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@AutoService(ComponentInstaller.class)
+public class MicrometerInstaller implements ComponentInstaller {
+  @Override
+  public void beforeByteBuddyAgent() {
+    SignalFxMeterRegistry signalFxRegistry =
+        new SignalFxMeterRegistry(new SplunkMetricsConfig(), Clock.SYSTEM);
+    NamingConvention signalFxNamingConvention = signalFxRegistry.config().namingConvention();
+    signalFxRegistry.config().namingConvention(new OtelNamingConvention(signalFxNamingConvention));
+
+    // TODO: agent does not respond?
+    try {
+      Logger errorLogger = LoggerFactory.getLogger(MicrometerInstaller.class);
+
+      Field exceptionListeners =
+          SignalFxMeterRegistry.class.getDeclaredField("onSendErrorHandlerCollection");
```

Comment:
there's no better way to do this? What does the built-in listener do right now?

Reply:
The built-in listener doesn't log the stacktrace. There's no setter or anything that allows providing your own listener, so I  had to force my way 🤷‍♂️